maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/DeriveSchemaMojo.java

```diff
@@ -0,0 +1,118 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.maven;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.confluent.kafka.schemaregistry.avro.AvroSchema;
+import io.confluent.kafka.schemaregistry.json.JsonSchema;
+import io.confluent.kafka.schemaregistry.maven.derive.schema.DeriveAvroSchema;
+import io.confluent.kafka.schemaregistry.maven.derive.schema.DeriveJsonSchema;
+import io.confluent.kafka.schemaregistry.maven.derive.schema.DeriveProtobufSchema;
+import io.confluent.kafka.schemaregistry.maven.derive.schema.DeriveSchema;
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema;
+import io.confluent.kafka.schemaregistry.utils.JacksonMapper;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.io.FileOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+@Mojo(name = "derive-schema", configurator = "custom-basic")
+public class DeriveSchemaMojo extends AbstractMojo {
+
+  @Parameter(required = true)
+  File messagePath;
+
+  @Parameter(required = true)
+  File outputPath;
+
+  @Parameter(defaultValue = "AVRO")
+  String schemaType;
+
+  protected static final ObjectMapper mapper = JacksonMapper.INSTANCE;
+
+  public static void writeOutput(File outputPath, JsonNode schemaInformation) throws IOException {
+    FileOutputStream fileStream = new FileOutputStream(outputPath.getPath());
+    OutputStreamWriter writer = new OutputStreamWriter(fileStream,
+        StandardCharsets.UTF_8);
+    writer.write(schemaInformation.toPrettyString());
+    writer.close();
+  }
+
+  public static List<JsonNode> readLinesOfMessages(File messagePath) throws IOException {
+    String content = MojoUtils.readFile(messagePath, StandardCharsets.UTF_8);
+    List<JsonNode> listOfMessages = new ArrayList<>();
+    String[] lineSeparatedMessages = content.split("\n");
+    for (int i = 0; i < lineSeparatedMessages.length; i++) {
+      // Ignore empty messages
+      if (lineSeparatedMessages[i].length() == 0) {
+        continue;
+      }
+      try {
+        listOfMessages.add(mapper.readValue(lineSeparatedMessages[i], ObjectNode.class));
+      } catch (IOException e) {
+        throw new IllegalArgumentException(
+            String.format("Message %d is an invalid JSON message", i), e);
```

Comment:
This message is not clear for the user. Either we should dump the entire invalid message or point out line number mentioning "message on line number x is invalid".

Reply:
The line number and message number will be the same. Each message is on a new line. 
For eg, If the file was:
```bash
{L":12}
{"G":1212}
{"G":1212, "DD":0.4}
```
We know message 0 on the first line is missing a ". The stack trace will look like:
```bash
Caused by: java.lang.IllegalArgumentException: Message 0 is an invalid JSON message
    at io.confluent.kafka.schemaregistry.maven.DeriveSchemaMojo.readLinesOfMessages (DeriveSchemaMojo.java:79)
    at io.confluent.kafka.schemaregistry.maven.DeriveSchemaMojo.execute (DeriveSchemaMojo.java:89)
...
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('L' (code 76)): was expecting double-quote to start field name
 at [Source: (String)"{L":12}"; line: 1, column: 3]
```
So the user knows which message is invalid and why

Reply:
A user will better understand message on line 1 is invalid rather than message 0 is invalid
there are two suggestions in this
1) index that starts with 1 and not 0
2) specifically naming it as line number.